lib/va_profile/profile_information/transaction_response.rb

```diff
@@ -0,0 +1,194 @@
+
+module VAProfile
+  module ProfileInformation
+    class TransactionResponse < VAProfile::Response
+      extend SentryLogging
+
+      attribute :transaction, VAProfile::Models::Transaction
+      ERROR_STATUS = 'COMPLETED_FAILURE'
+
+      attr_reader :response_body
+
+      def self.from(raw_response = nil, _user = nil)
+        @response_body = raw_response&.body
+        if error?
+          log_message_to_sentry(
+            'VAProfile transaction error',
+            :error,
+            { response_body: @response_body },
+            error: :va_profile
+          )
+        end
+
+        new(
+          raw_response&.status,
+          transaction: VAProfile::Models::Transaction.build_from(@response_body)
+        )
+      end
+
+      def self.error?
+        @response_body.try(:[], 'tx_status') == ERROR_STATUS
+      end
+    end
+
+    class AddressTransactionResponse < TransactionResponse
+      attribute :response_body, String
+
+      def self.from(*args)
+        return_val = super
+
+        log_error
+
+        return_val.response_body = @response_body
+        return_val
+      end
+
+      def changed_field
+        return :address unless response_body['tx_output']
+
+        address_pou = response_body['tx_output'][0]['address_pou']
+
+        case address_pou
+        when VAProfile::Models::BaseAddress::RESIDENCE
+          :residence_address
+        when VAProfile::Models::BaseAddress::CORRESPONDENCE
+          :correspondence_address
+        else
+          :address
+        end
+      end
+
+      def model_class
+        VAProfile::Models::Address
+      end
+
+      def self.log_error
+        if error?
+          PersonalInformationLog.create(
+            error_class: 'VAProfile::ProfileInformation::AddressTransactionResponseError',
+            data:
+              {
+                address: @response_body['tx_push_input'].except(
+                  'address_id',
+                  'originating_source_system',
+                  'source_system_user',
+                  'effective_start_date',
+                  'vet360_id'
+                ),
+                errors: @response_body['tx_messages']
+              }
+          )
+        end
+      rescue => e
+        log_exception_to_sentry(e)
+      end
+    end
+
+    class PersonTransactionResponse < TransactionResponse
+      NOT_FOUND_IN_MPI_CODE = 'MVI201'
+
+      def self.from(raw_response, user)
+        return_val = super(raw_response)
+        @user = user
+
+        log_mpi_error if @user.mpi_status == :ok
+
+        return_val
+      end
+
+      def model_class
+        VAProfile::Models::Person
+      end
+
+      def self.log_mpi_error
+        if error?
+          @response_body['tx_messages'].each do |tx_message|
+            if tx_message['code'] == NOT_FOUND_IN_MPI_CODE
+              return log_message_to_sentry(
+                'va profile mpi not found',
+                :error,
+                {
+                  icn: @user.icn,
```

Comment:
edipi and icn are considered PII... are these masked when logged to Sentry?

Reply:
Will look into this.